retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { InstallmentRepositry } from 'src/app/domain/installment/installment.repositry';
+import { Installment } from 'src/app/domain/installment/models/installment';
+
+@Component({
+  selector: 'app-installment',
+  templateUrl: './installment.component.html',
+  styleUrls: ['./installment.component.scss']
+})
+export class InstallmentComponent implements OnInit {
+  InstallmentList : Installment[] =[];
+  installForm!: FormGroup;
+  submit: boolean = false;
+  currentData!: Installment;
+  isButtonVisible: boolean = true;
+  displayedColumns: string[] = [
+    'id' ,
+    'installmentAmount',
+    'paymentAmount',
+    'dueDate',
+    'paymentDate',
+    'status',
+    'actions',
+  ];
+
+
+  constructor(private installmentrepositry : InstallmentRepositry , private formBuilder : FormBuilder) { }
+
+  ngOnInit(): void {
+    this.getAllInstallments();
+    this. InstallForm();
+  }
+
+  getAllInstallments(): void {
+    this.installmentrepositry.getList().subscribe((result) => {
+      this.InstallmentList = result;
+    });
+  }
+
+  DeleteInstallment(install :Installment) {
+    this.installmentrepositry.delete(install.id).subscribe(() => {
+      this.getAllInstallments();
+    });
+  }
+
+  InstallForm() {
+    this.installForm = this.formBuilder.group({
+      id: [''],
+      installmentAmount: [''],
+      paymentAmount: [''],
+      dueDate: ['', [Validators.required]],
+      paymentDate: [''],
+      status: [''],
```

Comment:
for now you can make installmentAmount and paymentAmount required

Reply:
Done